source/common/network/utility.cc

```diff
@@ -272,12 +273,17 @@ bool Utility::isLocalConnection(const Network::ConnectionSocket& socket) {
     }
 
     if (ifa->ifa_addr->sa_family == af_look_up) {
-      const auto* addr = reinterpret_cast<const struct sockaddr_storage*>(ifa->ifa_addr);
-      auto local_address = Address::addressFromSockAddr(
-          *addr, (af_look_up == AF_INET) ? sizeof(sockaddr_in) : sizeof(sockaddr_in6));
-
-      if (remote_address == local_address) {
-        return true;
+      if (af_look_up == AF_INET) {
+        const auto* addr = reinterpret_cast<const struct sockaddr_in*>(ifa->ifa_addr);
+        if (remote_address->ip()->ipv4()->address() == addr->sin_addr.s_addr) {
+          return true;
+        }
+      } else {
+        const auto* addr = reinterpret_cast<const struct sockaddr_in6*>(ifa->ifa_addr);
+        absl::uint128 v6addr = remote_address->ip()->ipv6()->address();
```

Comment:
nit: `const absl::uint128`

Reply:
might be good to call this `remote_v6addr` for maximum clarity, too.

Reply:
Done